ts/parsers/slideParser.ts

```diff
@@ -1,14 +1,141 @@
 import * as format from "string-template";
 import { getAttributeByPath, ZipHandler } from "../helpers";
-import { GraphicFrameParser, PowerpointElementParser } from "./";
+import { GraphicFrameParser, PowerpointElementParser, SlideRelationsParser } from "./";
 
 export default class SlideParser {
-    public static async getSlideElements(PPTElementParser: PowerpointElementParser, slideNumber): Promise<any[]> {
+    public static async getSlideLayout(slideRelations) {
+        // Read relationship filename of the slide (Get slideLayoutXX.xml)
+        // @sldFileName: ppt/slides/slide1.xml
+        // @resName: ppt/slides/_rels/slide1.xml.rels
+        let relationshipArray = slideRelations["Relationships"]["Relationship"];
+        let layoutFilename = "";
+        if (relationshipArray.constructor === Array) {
+            for (const relationship of relationshipArray) {
+                if (
+                    relationship["$"]["Type"] ===
+                    "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout"
+                ) {
+                    layoutFilename = relationship["$"]["Target"].replace("../", "ppt/");
+                    break;
+                }
+            }
+        } else {
+            layoutFilename = relationshipArray["$"]["Target"].replace("../", "ppt/");
+        }
+        // Open slideLayoutXX.xml
+        const slideLayoutContent = await ZipHandler.parseSlideAttributes(layoutFilename);
+
+        // Read slide master filename of the slidelayout (Get slideMasterXX.xml)
+        // @resName: ppt/slideLayouts/slideLayout1.xml
+        // @masterName: ppt/slideLayouts/_rels/slideLayout1.xml.rels
+        const slideLayoutResFilename =
+            layoutFilename.replace("slideLayouts/slideLayout", "slideLayouts/_rels/slideLayout") +
+            ".rels";
+        const slideLayoutResContent = await ZipHandler.parseSlideAttributes(slideLayoutResFilename);
+        relationshipArray = slideLayoutResContent["Relationships"]["Relationship"];
+        let masterFilename = "";
+        if (relationshipArray.constructor === Array) {
```

Comment:
check this as well if a cleaner array check can be done

Reply:
using `isArray` now